### ISSUE TITLE
Why is the prediction on trainset worse, although i trained it a long time?

### ISSUE THREAD

**Issue by thepate94227** [matterport/Mask_RCNN#1448]
Hello,

i use Mask-R-CNN to detect different ropes in images. I created my own dataset, which contains over 700 images. The shape of each image is (540,960).
I started to train all the layers from my NN and use many epoch (460), because i hope to get overfitting. So i can see on TensorBoard how many epochs is a good number before overfitting started. 
My config is:

```
GPU_COUNT = 1
IMAGES_PER_GPU = 2
NUM_CLASSES = 1 + 1  # background + 1 rope
BACKBONE = "resnet50"
IMAGE_MAX_DIM = 1024
IMAGE_MIN_DIM = 1024
RPN_ANCHOR_SCALES = rpn_anchor_scales
TRAIN_ROIS_PER_IMAGE = 32
STEPS_PER_EPOCH = 582
VALIDATION_STEPS = STEPS_PER_EPOCH / 6
USE_MINI_MASK = False
MAX_GT_INSTANCES = 15
DETECTION_MAX_INSTANCES = 15
LEARNING_RATE = 0.001
LEARNING_MOMENTUM = 0.9

model.train(dataset_train, dataset_validate, 
            learning_rate=config.LEARNING_RATE,
            epochs=460,
            layers="all",
            augmentation=augmentation)
```

My results are bad. Maybe it is because i didn't choose validation and testset very wisely... But for example this is a picture which the NN was trained by:
![04700](https://user-images.githubusercontent.com/5870642/56296192-4f663900-612e-11e9-8e0c-7b1bbb899977.png)

And this is an example of the testset:
![05040](https://user-images.githubusercontent.com/5870642/56296224-5bea9180-612e-11e9-8a97-265be292d3cd.png)

So after 460 epochs i saw the prediction and it didn't find any ropes. In the terminal there was written: *** No instances to display *** 

This is the Ground Truth, this has to be the result in the best case:
![gt2](https://user-images.githubusercontent.com/5870642/56296329-8fc5b700-612e-11e9-91a9-7c23877cb4f8.png)

One thing is weird: even if i use an image from the testset, i got bad result:
![train2](https://user-images.githubusercontent.com/5870642/56297644-fea40f80-6130-11e9-842a-2da1909c9480.png)

This is the result on TensorBoard:
![tensorboard2](https://user-images.githubusercontent.com/5870642/56298089-d668e080-6131-11e9-8b1f-03db3eac79e6.png)

You can see that the validation loss is oscillating, but more importantly the train loss is going down... But still the prediction is very bad, even if the image is from the trainset.

Does somebody know why my prediction of the train-image is very bad, although every epoch i went through the whole trainset and i choose 460 epochs? Are the number of epochs still to less? Or maybe some parameters like learning rate are too small or too high?

**Comment by KellyChenTue** [comment]
Hallo,
I think you are getting overfitting problem because your val loss is not going down consistently. That's why the prediction couldn't work as your expectation.
And I don't know why you mentioned you hope to get overfitting?

Your DETECTION_MIN_CONFIDENCE = 15, which makes it not possible at all to show any Mask.
As I know so far, it should be between 0 and 1. Maybe you can try to set it to 0.3 or 0.5 (means 30% confidence and 50% confidence respectively).

Correct me if I am wrong, thanks. :)

**Comment by thepate94227** [comment]
Thank you for your answer. I want overfitting because i save the weights after each epoch. I want to use the best weights, therfore i want the network to train and train and for example after 300 epochs its overfitted, but after 200 it has the best weights. 

My DETECTION_MIN_CONFIDENCE is not 15, my DETECTION_MAX_INSTANCES = 15.
